docs/reference/search/search-your-data/retrieve-selected-fields.asciidoc

```diff
@@ -6,72 +6,43 @@
 
 By default, each hit in the search response includes the document
 <<mapping-source-field,`_source`>>, which is the entire JSON object that was
-provided when indexing the document. To retrieve specific fields in the search
-response, you can use the `fields` parameter:
+provided when indexing the document. There are two recommended methods to
+retrieve selected fields from a search query:
 
-[source,console]
-----
-POST my-index-000001/_search
-{
-  "query": {
-    "match": {
-      "message": "foo"
-    }
-  },
-  "fields": ["user.id", "@timestamp"],
-  "_source": false
-}
-----
-// TEST[setup:my_index]
-
-The `fields` parameter consults both a document's `_source` and the index
-mappings to load and return values. Because it makes use of the mappings,
-`fields` has some advantages over referencing the `_source` directly: it
-accepts <<multi-fields, multi-fields>> and <<alias, field aliases>>, and
-also formats field values like dates in a consistent way.
-
-A document's `_source` is stored as a single field in Lucene. So the whole
-`_source` object must be loaded and parsed even if only a small number of
-fields are requested. To avoid this limitation, you can try another option for
-loading fields:
-
-* Use the <<docvalue-fields, `docvalue_fields`>>
-parameter to get values for selected fields. This can be a good
-choice when returning a fairly small number of fields that support doc values,
-such as keywords and dates.
-* Use the <<request-body-search-stored-fields, `stored_fields`>> parameter to
-get the values for specific stored fields (fields that use the
-<<mapping-store,`store`>> mapping option).
-
-You can also use the <<script-fields,`script_field`>> parameter to transform
-field values in the response using a script.
-
-You can find more detailed information on each of these methods in the
-following sections:
+* Use the <<search-fields-param,`fields` option>> to extract the values of
+fields present in the index mapping
+* Use the <<source-filtering,`_source` option>> if you need to access the original data that was passed at index time
 
-* <<search-fields-param>>
-* <<docvalue-fields>>
-* <<stored-fields>>
-* <<source-filtering>>
-* <<script-fields>>
+You can use both of these methods, though the `fields` option is preferred
+because it consults both the document `_source` and the index mappings to load
```

Comment:
I think this phrase "it consults both the document `_source`..." is slightly out-of-date and may cause confusion, since we load some field types from doc values, and also handle runtime fields (whose scripts may be entirely based on doc values). Maybe we could say "it consults both the document data and index mappings..." ?

If we make that change, then in the next section it'd be helpful to mention that for standard fields, we always attempt to load from _source. Adding a sentence like that will make it clear that `fields` expects all standard fields to be present in _source, and has the same drawback of _source filtering where we need to load and parse the whole source to retrieve just one field.

Reply:
++ Thanks for these clarifications @jtibshirani!

I'll change that sentence to read:
>You can use both of these methods, though the `fields` option is preferred because it consults both the document data and index mappings.

To address your second point, does this description accurately describe the behavior?
>When retrieving standard fields using the `fields` option, {es} always attempts to load values from `_source`. This behavior has the same implications of source filtering where {es} needs to load and parse the entire `_source` to
retrieve just one field.

Reply:
`fields` presents values in the way ES sees them. For most values that is exactly how they are indexed. For `text` its just the text from the `_source`, but other values come back in other ways. Right now they do mostly read from `_source` and "normalize" to look like they were indexed but that isn't part of the contract - we could decide at some point to get them from doc values or stored fields or some new thing that we have yet to invent. So I'd try to say *how* we do it, just what we do.

Reply:
That's a good point @nik9000, always loading from _source definitely isn't part of the contract. I do think it's helpful though to give some sense how the option currently works, to help users build a simple mental model. Maybe we could use phrasing like "The current behavior is to ...", or "Currently, the `fields` option is implemented by..." to make it clear we're not defining the API contract here?

I think we can avoid mentioning "... where {es} needs to load and parse the entire _source to retrieve just one field" in this prominent location, since it is very rarely an issue! We could save that note for the "Other methods of retrieving data" you created, so that only users interested in digging further will see it.

Reply:
Thank you both for the context! Does something like this statement encapsulate both of your points?

>When retrieving standard fields, the `fields` option returns values in the way that {es} sees them. For most field types, this means that the `fields` option returns values from `_source` exactly how they are indexed.